core/src/avm2/array.rs

```diff
@@ -85,44 +291,171 @@ impl<'gc> ArrayStorage<'gc> {
     ///
     /// Holes are copied as holes and not resolved at append time.
     pub fn append(&mut self, other_array: &Self) {
-        for other_item in other_array.storage.iter() {
-            self.storage.push(*other_item)
+        match self {
+            ArrayStorage::Dense {
+                storage,
+                occupied_count,
+            } => match &other_array {
+                ArrayStorage::Dense {
```

Comment:
could we be generic here? As in, just "for each item of `other_array`, add the item to our storage" - using either indices and `get`, or the iterator?